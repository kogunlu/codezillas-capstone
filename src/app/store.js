import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import bookingReducer from '../features/booking/bookingSlice';
import userReducer from '../features/user/userSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    booking: bookingReducer,
    user: userReducer,
  },
});
