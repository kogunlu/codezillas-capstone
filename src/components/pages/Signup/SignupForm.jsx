/* eslint-disable prefer-destructuring */
/* eslint-disable no-unused-vars */
import React, { useEffect, useState } from 'react';
import swal from 'sweetalert';
import { getAuth, createUserWithEmailAndPassword } from 'firebase/auth';
import { doc, setDoc, getDoc } from 'firebase/firestore';
import { useNavigate } from 'react-router-dom';
import { useFormik } from 'formik';
import db from '../../../db/firebase.config';
import userPicture from './Signup_picture/user.jpg';

function SignupForm() {
  const navigate = useNavigate();

  const validate = (values) => {
    const errors = {};

    if (!values.firstName) {
      errors.firstName = '*Required';
    } else if (values.firstName.length > 15) {
      errors.firstName = 'Must be 15 characters or less';
    }

    if (!values.lastName) {
      errors.lastName = '*Required';
    } else if (values.lastName.length > 20) {
      errors.lastName = 'Must be 20 characters or less';
    }

    if (!values.email) {
      errors.email = '*Required';
    } else if (
      !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)
    ) {
      errors.email = 'Invalid email address';
    }

    if (!values.confirmEmail) {
      errors.confirmEmail = '*Required';
    } else if (values.email !== values.confirmEmail) {
      errors.confirmEmail = 'Email addresses do not match.';
    }

    if (!values.password) {
      errors.password = '*Required';
    } else if (values.password.length < 6) {
      errors.password = 'Must be minimum of 6 characters.';
    }

    if (!values.confirmPassword) {
      errors.confirmPassword = '*Required';
    } else if (values.password !== values.confirmPassword) {
      errors.confirmPassword = 'Passwords do not match.';
    }

    if (!values.confirmPassword) {
      errors.confirmPassword = '*Required';
    } else if (values.password !== values.confirmPassword) {
      errors.confirmPassword = 'Passwords do not match.';
    }

    if (!values.day) {
      errors.day = '*Required';
    } else if (values.day < 1 || values.day > 31) {
      errors.day = 'Please submit a valid day.';
    } else if (
      (values.month === 2 && (values.day < 1 || values.day > 29)) ||
      ((values.month === 4 ||
        values.month === 6 ||
        values.month === 9 ||
        values.month === 11) &&
        (values.day < 1 || values.day > 30))
    ) {
      errors.day = 'Please submit a valid day.';
    }

    if (!values.month) {
      errors.month = '*Required';
    } else if (values.month < 1 || values.month > 12) {
      errors.month = 'Please submit a valid month.';
    }

    if (!values.year) {
      errors.year = '*Required';
    } else if (values.year < 1900 || values.year > 2023) {
      errors.year = 'You can`t be that old!';
    }

    return errors;
  };

  const formik = useFormik({
    initialValues: {
      firstName: '',
      lastName: '',
      email: '',
      confirmEmail: '',
      password: '',
      confirmPassword: '',
      day: '',
      month: '',
      year: '',
    },
    validate,
    onSubmit: async (values) => {
      const docRef = doc(db, 'user-details', formik.values.email);
      const docSnap = await getDoc(docRef);

      if (docSnap.exists()) {
        swal({
          title: 'Opps!',
          text: 'You already have an account, please try to log in!',
          icon: 'error',
        });
      } else {
        const auth = getAuth();

        const displayName = formik.values.firstName;
        const email = formik.values.email;
        const password = formik.values.password;
        const birthdate = formik.values.birthdate;

        createUserWithEmailAndPassword(auth, email, password)
          .then(
            setDoc(doc(db, 'user-details', formik.values.email), {
              name: `${formik.values.firstName} ${formik.values.lastName}`,
              email: formik.values.email,
              birthdate: `${formik.values.day}-${formik.values.month}-${formik.values.year}`,
              photoUrl: userPicture,
            })
          )
          .then((userCredential) => {
            const { user } = userCredential.user;

            navigate('/thanks-signup');
          })
          .catch((error) => {
            const errorCode = error.code;
            const errorMessage = error.message;

            swal('Error!', 'Something went wrong, try again.', 'error');
          });
      }
    },
  });

  function handleLoginClick() {
    navigate('/login');
  }

  return (
    <form
      className="w-full h-full py-10 flex flex-col justify-between gap-5 items-center px-2 md:px-5"
      onSubmit={formik.handleSubmit}
    >
      <div className="w-full h-20 flex justify-between">
        <div className="w-5/12 flex flex-col">
          <input
            type="text"
            className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
            placeholder="First Name"
            id="firstName"
            {...formik.getFieldProps('firstName')}
          />
          {formik.touched.firstName && formik.errors.firstName ? (
            <div className="text-red-400 italic text-sm lg:text-base">
              {formik.errors.firstName}
            </div>
          ) : null}
        </div>

        <div className="w-5/12 flex flex-col">
          <input
            type="text"
            className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
            placeholder="Last Name"
            id="lastName"
            {...formik.getFieldProps('lastName')}
          />
          {formik.touched.lastName && formik.errors.lastName ? (
            <div className="text-red-400 italic text-sm lg:text-base">
              {formik.errors.lastName}
            </div>
          ) : null}
        </div>
      </div>

      <div className="w-full h-20 flex flex-col">
        <input
          type="email"
          className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
          placeholder="Your Email"
          id="email"
          {...formik.getFieldProps('email')}
        />
        {formik.touched.email && formik.errors.email ? (
          <div className="text-red-400 italic text-sm lg:text-base">
            {formik.errors.email}
          </div>
        ) : null}
      </div>

      <div className="w-full h-20 flex flex-col">
        <input
          type="email"
          className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
          placeholder="Confirm email"
          id="confirmEmail"
          {...formik.getFieldProps('confirmEmail')}
        />
        {formik.touched.confirmEmail && formik.errors.confirmEmail ? (
          <div className="text-red-400 italic text-sm lg:text-base">
            {formik.errors.confirmEmail}
          </div>
        ) : null}
      </div>

      <div className="w-full h-20 flex justify-between">
        <div className="w-5/12 flex flex-col">
          <input
            type="password"
            className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
            placeholder="Password"
            id="password"
            {...formik.getFieldProps('password')}
          />
          {formik.touched.password && formik.errors.password ? (
            <div className="text-red-400 italic text-sm lg:text-base">
              {formik.errors.password}
            </div>
          ) : null}
        </div>

        <div className="w-5/12 flex flex-col">
          <input
            type="password"
            className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
            placeholder="Confirm Password"
            id="confirmPassword"
            {...formik.getFieldProps('confirmPassword')}
          />
          {formik.touched.confirmPassword && formik.errors.confirmPassword ? (
            <div className="text-red-400 italic text-sm lg:text-base">
              {formik.errors.confirmPassword}
            </div>
          ) : null}
        </div>
      </div>

      <div className="w-full h-24 flex flex-col justify-around">
        <div className="w-full flex justify-between">
          <div className="w-5/12 flex justify-between">
            <p
              htmlFor="birthdate"
              className="text-gray-400 w-5/12 lg:w-7/12 text-center lg:pt-5"
            >
              Birth Date
            </p>
            <input
              id="day"
              name="day"
              type="number"
              className="border w-3/12 h-12 md:h-16 rounded-md text-center lg:text-start lg:pl-3 focus:outline-none focus:shadow-lg"
              placeholder="DD"
              {...formik.getFieldProps('day')}
            />
          </div>

          <div className="w-5/12 flex justify-between">
            <input
              id="month"
              name="month"
              type="number"
              className="border w-3/12 h-12 md:h-16 rounded-md text-center lg:text-start lg:pl-2 focus:outline-none focus:shadow-lg"
              placeholder="MM"
              {...formik.getFieldProps('month')}
            />

            <input
              id="year"
              name="year"
              type="number"
              className="border w-7/12 h-12 md:h-16 rounded-md text-center lg:text-start lg:pl-3 lg:px-5 focus:outline-none focus:shadow-lg"
              placeholder="YYYY"
              {...formik.getFieldProps('year')}
            />
          </div>
        </div>

        <div className="w-full h-8 flex justify-between">
          <div className="w-5/12 flex justify-end">
            {formik.touched.day && formik.errors.day ? (
              <div className="text-red-400 text-sm lg:text-base italic">
                {formik.errors.day}
              </div>
            ) : null}
          </div>

          <div className="w-5/12 flex justify-between">
            <div className="w-full flex justify-start ">
              {formik.touched.month && formik.errors.month ? (
                <div className="text-red-400 text-sm lg:text-base italic	">
                  {formik.errors.month}
                </div>
              ) : null}
            </div>

            <div className="w-full flex justify-end">
              {formik.touched.year && formik.errors.year ? (
                <div className="text-red-400 text-sm lg:text-base italic 	">
                  {formik.errors.year}
                </div>
              ) : null}
            </div>
          </div>
        </div>
      </div>

      <div className="w-full flex justify-between items-center">
        <input
          type="button"
          value="Login"
          className="py-1 md:py-3 px-5 h-10 md:h-14 hover:bg-cyan-200 focus:outline-none hover:scale-105 border border-cyan-400 bg-cyan-400 text-lg font-semibold rounded-md w-5/12 shadow-lg"
          onClick={() => handleLoginClick()}
        />

        <input
          type="submit"
          value="Signup"
          className=" py-1 md:py-3 px-5 h-10 md:h-14 focus:outline-none hover:scale-105 hover:bg-cyan-400 hover:text-white text-cyan-400 border border-cyan-400 bg-white text-lg font-semibold rounded-md w-5/12 "
        />
      </div>
    </form>
  );
}

export default SignupForm;
