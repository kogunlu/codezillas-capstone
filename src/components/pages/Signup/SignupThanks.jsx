import React from 'react';
import Thanks from '../../shared/thanksPages/Thanks';

function SignupThanks() {
  return (
    <div>
      <Thanks text="Your Sign Up request has been received, you will soon receive a confirmation email. Please follow the steps in the email to complete and activate your account." />
    </div>
  );
}

export default SignupThanks;
