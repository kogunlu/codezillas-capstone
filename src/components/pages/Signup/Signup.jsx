/* eslint-disable prefer-destructuring */
/* eslint-disable no-unused-vars */
import React from 'react';
import Socials from '../Login/Socials';
import signupPicture from './Signup_picture/signup.jpg';
import SignupForm from './SignupForm';

function Signup() {
  return (
    <div className="h-full w-full flex flex-col justify-center items-center lg:items-end gap-5 mt-10 mb-5">
      <div className="w-6/12 lg:mr-5 xl:mr-8 2xl:mr-12">
        <h2 className="text-center lg:text-start text-2xl font-semibold">
          SIGNUP NOW
        </h2>
      </div>

      <div className=" h-full w-full flex justify-center items-start md:items-center">
        <div className="h-4/6 md:h-full w-full lg:w-11/12 flex flex-col items-start justify-start md:justify-between">
          <div className="flex flex-col gap-5 lg:gap-0 lg:flex-row h-min w-full justify-evenly items-center">
            <div className="w-full md:w-10/12 lg:w-5/12 h-min flex items-center justify-center md:items-start">
              <img src={signupPicture} alt="login_picture" />
            </div>

            <div className="h-auto md:h-4/6 w-full md:w-10/12 lg:w-6/12 flex flex-col justify-evenly md:justify-center md:gap-5 lg:justify-evenly lg:gap-0 items-center">
              <div className="w-full h-max md:h-3/6 lg:h-full shadow-xl rounded-md flex justify-center items-center ">
                <SignupForm />
              </div>

              <div className="w-10/12 flex flex-col items-center justify-center">
                <div className="w-full flex items-center justify-evenly">
                  <div className="w-5/12">
                    <hr className="border-1 border-cyan-400" />
                  </div>

                  <p className="w-2/12 mt-2 text-center text-base xl:text-xl text-slate-600">
                    Or
                  </p>

                  <div className="w-5/12">
                    <hr className="border-1 border-cyan-400" />
                  </div>
                </div>

                <Socials />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Signup;
