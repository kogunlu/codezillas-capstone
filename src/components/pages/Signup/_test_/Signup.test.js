import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { store } from '../../../../app/store';
import Signup from '../Signup';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

it('renders correctly', async () => {
  const errorComponent = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Signup />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(errorComponent).toMatchSnapshot();
});
