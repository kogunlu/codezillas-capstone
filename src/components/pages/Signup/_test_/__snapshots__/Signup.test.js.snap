// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="h-full w-full flex flex-col justify-center items-center lg:items-end gap-5 mt-10 mb-5"
>
  <div
    className="w-6/12 lg:mr-5 xl:mr-8 2xl:mr-12"
  >
    <h2
      className="text-center lg:text-start text-2xl font-semibold"
    >
      SIGNUP NOW
    </h2>
  </div>
  <div
    className=" h-full w-full flex justify-center items-start md:items-center"
  >
    <div
      className="h-4/6 md:h-full w-full lg:w-11/12 flex flex-col items-start justify-start md:justify-between"
    >
      <div
        className="flex flex-col gap-5 lg:gap-0 lg:flex-row h-min w-full justify-evenly items-center"
      >
        <div
          className="w-full md:w-10/12 lg:w-5/12 h-min flex items-center justify-center md:items-start"
        >
          <img
            alt="login_picture"
            src="signup.jpg"
          />
        </div>
        <div
          className="h-auto md:h-4/6 w-full md:w-10/12 lg:w-6/12 flex flex-col justify-evenly md:justify-center md:gap-5 lg:justify-evenly lg:gap-0 items-center"
        >
          <div
            className="w-full h-max md:h-3/6 lg:h-full shadow-xl rounded-md flex justify-center items-center "
          >
            <form
              className="w-full h-full py-10 flex flex-col justify-between gap-5 items-center px-2 md:px-5"
              onSubmit={[Function]}
            >
              <div
                className="w-full h-20 flex justify-between"
              >
                <div
                  className="w-5/12 flex flex-col"
                >
                  <input
                    className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                    id="firstName"
                    name="firstName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="First Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="w-5/12 flex flex-col"
                >
                  <input
                    className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                    id="lastName"
                    name="lastName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Last Name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="w-full h-20 flex flex-col"
              >
                <input
                  className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                  id="email"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Your Email"
                  type="email"
                  value=""
                />
              </div>
              <div
                className="w-full h-20 flex flex-col"
              >
                <input
                  className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                  id="confirmEmail"
                  name="confirmEmail"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Confirm email"
                  type="email"
                  value=""
                />
              </div>
              <div
                className="w-full h-20 flex justify-between"
              >
                <div
                  className="w-5/12 flex flex-col"
                >
                  <input
                    className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                    id="password"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="w-5/12 flex flex-col"
                >
                  <input
                    className="border w-full h-12 md:h-16 rounded-md pl-2 px-5 focus:outline-none focus:shadow-lg text-sm lg:text-base"
                    id="confirmPassword"
                    name="confirmPassword"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    type="password"
                    value=""
                  />
                </div>
              </div>
              <div
                className="w-full h-24 flex flex-col justify-around"
              >
                <div
                  className="w-full flex justify-between"
                >
                  <div
                    className="w-5/12 flex justify-between"
                  >
                    <p
                      className="text-gray-400 w-5/12 lg:w-7/12 text-center lg:pt-5"
                      htmlFor="birthdate"
                    >
                      Birth Date
                    </p>
                    <input
                      className="border w-3/12 h-12 md:h-16 rounded-md text-center lg:text-start lg:pl-3 focus:outline-none focus:shadow-lg"
                      id="day"
                      name="day"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="DD"
                      type="number"
                      value=""
                    />
                  </div>
                  <div
                    className="w-5/12 flex justify-between"
                  >
                    <input
                      className="border w-3/12 h-12 md:h-16 rounded-md text-center lg:text-start lg:pl-2 focus:outline-none focus:shadow-lg"
                      id="month"
                      name="month"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="MM"
                      type="number"
                      value=""
                    />
                    <input
                      className="border w-7/12 h-12 md:h-16 rounded-md text-center lg:text-start lg:pl-3 lg:px-5 focus:outline-none focus:shadow-lg"
                      id="year"
                      name="year"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="YYYY"
                      type="number"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="w-full h-8 flex justify-between"
                >
                  <div
                    className="w-5/12 flex justify-end"
                  />
                  <div
                    className="w-5/12 flex justify-between"
                  >
                    <div
                      className="w-full flex justify-start "
                    />
                    <div
                      className="w-full flex justify-end"
                    />
                  </div>
                </div>
              </div>
              <div
                className="w-full flex justify-between items-center"
              >
                <input
                  className="py-1 md:py-3 px-5 h-10 md:h-14 hover:bg-cyan-200 focus:outline-none hover:scale-105 border border-cyan-400 bg-cyan-400 text-lg font-semibold rounded-md w-5/12 shadow-lg"
                  onClick={[Function]}
                  type="button"
                  value="Login"
                />
                <input
                  className=" py-1 md:py-3 px-5 h-10 md:h-14 focus:outline-none hover:scale-105 hover:bg-cyan-400 hover:text-white text-cyan-400 border border-cyan-400 bg-white text-lg font-semibold rounded-md w-5/12 "
                  type="submit"
                  value="Signup"
                />
              </div>
            </form>
          </div>
          <div
            className="w-10/12 flex flex-col items-center justify-center"
          >
            <div
              className="w-full flex items-center justify-evenly"
            >
              <div
                className="w-5/12"
              >
                <hr
                  className="border-1 border-cyan-400"
                />
              </div>
              <p
                className="w-2/12 mt-2 text-center text-base xl:text-xl text-slate-600"
              >
                Or
              </p>
              <div
                className="w-5/12"
              >
                <hr
                  className="border-1 border-cyan-400"
                />
              </div>
            </div>
            <div
              className="w-full flex items-center justify-center gap-16"
            >
              <button
                className="w-1/12 flex justify-end"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="text-cyan-400 text-2xl xl:text-3xl hover:text-cyan-200"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                  />
                </svg>
              </button>
              <button
                className="w-1/12 flex justify-start"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="text-cyan-400 text-2xl xl:text-3xl hover:text-cyan-200"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.26620003,9.76452941 C6.19878754,6.93863203 8.85444915,4.90909091 12,4.90909091 C13.6909091,4.90909091 15.2181818,5.50909091 16.4181818,6.49090909 L19.9090909,3 C17.7818182,1.14545455 15.0545455,0 12,0 C7.27006974,0 3.1977497,2.69829785 1.23999023,6.65002441 L5.26620003,9.76452941 Z"
                  />
                  <path
                    d="M16.0407269,18.0125889 C14.9509167,18.7163016 13.5660892,19.0909091 12,19.0909091 C8.86648613,19.0909091 6.21911939,17.076871 5.27698177,14.2678769 L1.23746264,17.3349879 C3.19279051,21.2936293 7.26500293,24 12,24 C14.9328362,24 17.7353462,22.9573905 19.834192,20.9995801 L16.0407269,18.0125889 Z"
                  />
                  <path
                    d="M19.834192,20.9995801 C22.0291676,18.9520994 23.4545455,15.903663 23.4545455,12 C23.4545455,11.2909091 23.3454545,10.5272727 23.1818182,9.81818182 L12,9.81818182 L12,14.4545455 L18.4363636,14.4545455 C18.1187732,16.013626 17.2662994,17.2212117 16.0407269,18.0125889 L19.834192,20.9995801 Z"
                  />
                  <path
                    d="M5.27698177,14.2678769 C5.03832634,13.556323 4.90909091,12.7937589 4.90909091,12 C4.90909091,11.2182781 5.03443647,10.4668121 5.26620003,9.76452941 L1.23999023,6.65002441 C0.43658717,8.26043162 0,10.0753848 0,12 C0,13.9195484 0.444780743,15.7301709 1.23746264,17.3349879 L5.27698177,14.2678769 Z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
